ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/Types.hs

```diff
@@ -1026,3 +1057,26 @@ data TraceIteratorEvent blk
 newtype TraceChainSelStarvationEvent blk
   = ChainSelStarvation (Enclosing' (RealPoint blk))
   deriving (Generic, Eq, Show)
+
+data TraceAddPerasCertEvent blk
+  = -- | The Peras certificate from the given round boosting the given block was
+    -- added to the queue. The size of the queue is included.
+    AddedPerasCertToQueue PerasRoundNo (Point blk) (Enclosing' Word)
+  | -- | The Peras certificate from the given round boosting the given block was
+    -- popped from the queue.
+    PoppedPerasCertFromQueue PerasRoundNo (Point blk)
+  | -- | The Peras certificate from the given round boosting the given block was
+    -- too old, ie its slot was older than the current immutable slot (the third
+    -- argument).
```

Comment:
Maybe we want to wrap the the `Point blk` arguments in a newtype to clarify the thing each one denotes, eg `(CurrentImmutableSlot blk)`, where

```haskell
newtype CurrentImmutableSlot blk = CurrentImmutableSlot (Point blk) ...
```

But this might be too heavy handed. I just worry the trace messages might be confusing otherwise.